tests/support/util.tcl

```diff
@@ -681,3 +681,29 @@ proc string2printable s {
     set res "\"$res\""
     return $res
 }
+
+proc test_histogram_distribution {res min_prop max_prop} {
+    # 1) Check that probability of each element are between {min_prop} and {max_prop}.
+    unset -nocomplain mydict
+    foreach key $res {
+        dict incr mydict $key 1
+    }
+
+    foreach key [dict keys $mydict] {
+        set value [dict get $mydict $key]
+        set probability [expr {double($value) / [llength $res]}]
+        assert {$probability > $min_prop && $probability < $max_prop}
+    }
+
+    # 2) Check order is not completely continuous.
+    set firstkey [lindex $res 0]
+    set identical 1
+    set check_count [expr {[llength $res] / 10}];
+    foreach key [lrange $res 1 $check_count] {
+        if {$key != $firstkey} {
+            set identical 0
+            break
+        }
+    }
+    assert {$identical == 0}
```

Comment:
i wanna use this function for checking random distribution in the cases that don't offer random order (positive count argument). so maybe this check should be extracted to a function of it's own.

however, come to think of it:
1. this entire function is not so easy to test for the other cases (the ones that do offer uniqueness), since they'll not provide repetition (unless repeatedly called and appending the results to a list, but then this part of the test doesn't have a lot of value)
2. the way to test random order without repetition is maybe to see that the order changes between calls. i.e. that one time we get `[2, 5]` and other times we get `[5, 2]`.

bottom line, i'm leaning towards dropping that check entirely (the bottom half of this function). it's not very generic, and doesn't have a high value

Reply:
Just now, after your reminder above, I verified that the continuity of log(n)(100000) at the beginning is probably valid. Or calculate the maximum number of consecutive times for the entire list.

Reply:
i don't understand what you mean. but again, i think we should drop the entire bottom part of this function. i don't think it gives much value and it's not generic enough.

then, i'd like to use that function to test random distribution for the positive count variant.
i'd suggest to move the new test line you added to be near: `# PATH 1: Use negative count.`
and then add a similar test inside `foreach size {8 2} {` to test `PATH 3` and `PATH 4`
for these, you'll need to accumulate the results of the iterations into a list (current code puts them in a dict)